Automattic/jetpack
# Gutenberg: Create new StripeConnectToolbarButton shared component

Fixes https://github.com/Automattic/jetpack/issues/23117

#### Changes proposed in this Pull Request:

* Create a `StripeConnectToolbarButton` shared component, reusable across several payment blocks.
* Add the `StripeConnectToolbarButton` component to the Premium Content and Payment Button blocks.
* Add the `jetpack_editor_block_stripe_connect_click` Tracks event to the `StripeConnectToolbarButton` button.

#### Jetpack product discussion

pdfrp0-jB-p2

#### Does this pull request change what data or activity we track or use?

No.
The event added to the button is the same already used in the `StripeNudge` banner for the exact same purpose.

#### Testing instructions:






* Sync these changes to WPCOM.
* Sandbox a site with a premium or business plan without a pre-existing Stripe connection.
* Enter `localStorage.setItem( 'debug', 'dops:analytics' );` in the console and reload.
* Insert a Payment Button or Premium Content block.
* Make sure the "Connect Stripe" button in the block toolbar is visible and doesn't introduce regressions.
* Click on it.
* Check the console and make sure the`jetpack_editor_block_stripe_connect_click` is recorded (you might need to enable "preserve log" since the button will navigate away.
* Complete the connection; you'll be redirected to the editor.
* Make sure the "Connect Stripe" button is not rendered in the block toolbar anymore.
